contracts/ReversibleICO.sol

```diff
@@ -1031,7 +1031,7 @@ contract ReversibleICO is IERC777Recipient {
 
                         // update return values
                         returnedETH = returnedETH.add(returnValue);
-                        participantRecord.returnedETH = participantRecord.returnedETH.add(returnValue);
+                        participantRecord.byStage[0].returnedETH = participantRecord.byStage[0].returnedETH.add(returnValue);
```

Comment:
same as above

and where we use byStage in a loop just assign from ` byStage 
 = participantsByAddress[_participantAddress].byStage[i]; `